src/store/utilities/mergeFloorplans.js

```diff
@@ -0,0 +1,210 @@
+      id: `${side}${window.id}`,
+      window_definition_id: `${side}${window.window_definition_id}`,
+    }));
+
+    const doors = story.doors.map(door => ({
+      ...door,
+      edge_id: `${side}${door.edge_id}`,
+      id: `${side}${door.id}`,
+      door_definition_id: `${side}${door.door_definition_id}`,
+    }));
+
+    const shading = story.shading.map(shade => ({
+      ...shade,
+      id: `${side}${shade.id}`,
+      face_id: `${side}${shade.face_id}`,
+    }));
+
+    const images = story.images.map(image => ({
+      ...image,
+      id: `${side}${image.id}`,
+    }));
+
+    return {
+      ...story,
+      id: `${side}${story.id}`,
+      geometry,
+      spaces,
+      windows,
+      doors,
+      shading,
+      images,
+    };
+  });
+
+  // CURRENT WINDOW DEFS
+  data.window_definitions = data.window_definitions.map(window_def => ({
+    ...window_def,
+    id: `${side}${window_def.id}`,
+  }));
+
+  // CURRENT DOOR DEFS
+  data.door_definitions = data.door_definitions.map(door_def => ({
+    ...door_def,
+    id: `${side}${door_def.id}`,
+  }));
+
+  // CURRENT DAYLIGHTING DEFS
+  data.daylighting_control_definitions = data.daylighting_control_definitions.map(control => ({
+    ...control,
+    id: `${side}${control.id}`,
+  }));
+
+  // CURRENT THERMAL ZONES
+  data.thermal_zones = data.thermal_zones.map(zone => ({
+    ...zone,
+    id: `${side}${zone.id}`,
+  }));
+
+  // CURRENT BUILDING UNITS
+  data.building_units = data.building_units.map(unit => ({
+    ...unit,
+    id: `${side}${unit.id}`,
+  }));
+
+  // CURRENT SPACE TYPES
+  data.space_types = data.space_types.map(space => ({
+    ...space,
+    id: `${side}${space.id}`,
+  }));
+
+  // CURRENT PITCHED ROOFS
+  data.pitched_roofs = data.pitched_roofs.map(roof => ({
+    ...roof,
+    id: `${side}${roof.id}`,
+  }));
+
+  return data;
+}
+
+/**
+ * Takes in the current state to get the current floorplan
+ * Generates like structure for both current and imported floorplan
+ * Merges them into one object and imports that as 1 new floorplan
+ *
+ * @param {Context} context
+ * @param {Payload} payload
+ */
+export default function mergeFloorplans(context, payload) {
+  // PREP FLOORPLANS FOR MERGE
+  const currentFloorplan = exportData(context.state, context.getters);
+
+  const prepedCurrentFloorplan = prepareDataForMerge('l', currentFloorplan);
+  const prepedIncomingFloorplan = prepareDataForMerge('r', payload.data);
+
+  // MERGE FLOORPLANS PROPERTIES
+  const zipUpStories = _.zip(prepedIncomingFloorplan.stories, prepedCurrentFloorplan.stories);
+  const mergeStories = zipUpStories.map((pairOfStories) => {
+    if (!pairOfStories[0]) {
+      return pairOfStories[1];
+    }
+    if (!pairOfStories[1]) {
+      return pairOfStories[0];
+    }
+    return {
+      ...pairOfStories[0],
+      geometry: {
+        id: pairOfStories[0].id,
+        edges: [...pairOfStories[0].geometry.edges, ...pairOfStories[1].geometry.edges],
```

Comment:
This seems like it might be a good place for destructuring. `leftStory` and `rightStory` might be easier to read than `pairOfStories[0]` and `pairOfStories[1]`

Reply:
Agreed - updated this.